pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -313,6 +313,78 @@ self: super: {
     '';
   });
 
+  coq_nvim = super.coq_nvim.overrideAttrs (old: {
+    passthru.python3Dependencies = let
+
+      # Extract the version of the dependencies from the requirements.txt file
+      requirements = builtins.readFile "${old.src}/requirements.txt";
+      requirementsLines = lib.splitString "\n" requirements;
+
+      extractLine = pname: lib.findSingle
+        (lib.strings.hasPrefix pname)
+        "none"
+        "multiple"
+        requirementsLines;
+
+      extractDepVersion = pname:
+        let
+          line = extractLine pname;
+          hashPart = lib.lists.last (lib.splitString "/" line);
+          hash = lib.strings.removeSuffix ".tar.gz" hashPart;
+        in hash;
+    in ps: with ps; [
+      pynvim
+      pyyaml
+      (let
+        version = extractDepVersion "pynvim_pp";
+      in buildPythonPackage {
+        pname = "pynvim_pp";
+        inherit version;
+        format = "pyproject";
+        propagatedBuildInputs = [ setuptools pynvim ];
+        src = fetchFromGitHub {
+          owner = "ms-jpq";
+          repo = "pynvim_pp";
+          rev = version;
+          fetchSubmodules = false;
+
+          # This maps the current version of pynvim_pp to the sha256 of its source.
+          # Hence, if the dependency version changes in the requirements.txt, the build will fail.
+          hash = { "456765f6dc8cef6df39ae3e61c79585258d38517" = "sha256-edDDzxR60ILFbv1CnYGxW/sJDMeRIFPjmTeiMRdvA3k="; }.${version};
+        };
+      })
+      (let
+        version = extractDepVersion "std2";
+      in buildPythonPackage {
+        pname = "std2";
+        inherit version;
+        format = "pyproject";
+        propagatedBuildInputs = [ setuptools ];
+        src = fetchFromGitHub {
+          owner = "ms-jpq";
+          repo = "std2";
+          rev = version;
+          fetchSubmodules = false;
+
+          # This maps the current version of std2 to the sha256 of its source.
+          # Hence, if the dependency version changes in the requirements.txt, the build will fail.
+          hash = { "963cf22346620926c0bd64f628ff4d8141123db5" = "sha256-drW7eZKE/NmVpkZfiA7nRlgUeqNNDnKA9h1qVADDZ/s="; }.${version};
+        };
+        doCheck = true;
+      })
+    ];
+
+    # We need some patches so it stops complaining about not being in a venv
+    postPatch = ''
+      substituteInPlace coq/consts.py \
+        --replace "VARS = TOP_LEVEL / \".vars\"" "VARS = Path.home() / \".cache/coq_nvim/vars\"";
+      substituteInPlace coq/__main__.py \
+        --replace "_IN_VENV = _RT_PY == _EXEC_PATH" "_IN_VENV = True"
+      substituteInPlace coq/__main__.py \
+        --replace "elif lock != _REQ:" "elif False:"
+    '';
+  });
```

Comment:
Would prefer this to be a proper patch instead of using `substituteInPlace`. This is the kinda of think that if upstream changes we need to fix the patch.

Reply:
Done !